packages/rum-core/src/common/apm-server.js

```diff
@@ -150,7 +155,20 @@ class ApmServer {
         const { status, responseText } = xhr
         reject({ url, status, responseText })
       }
-      xhr.send(payload)
+
+      let canSend = true
+      if (typeof beforeSend === 'function') {
+        canSend = beforeSend(xhr, url, method, headers, payload)
```

Comment:
Also let's pass those params as object if we are going with this approach. Too many args

Reply:
That's a good idea, will do that.